pulsar/log/wrapper_slog.go

```diff
@@ -0,0 +1,114 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package log
```

Comment:
```suggestion

//go:build go1.21

package log
```

I suggest you use Go build constraints. When you use go1.21+, this file will be compiled, otherwise, not.

This can keep us compatible with the go 1.20.

Reply:
This is an elegant and incredible suggestion, I knew about build constrains, but I never had the need to use them. 
Doing it, and reverting the go version upgrade.